src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -43,17 +46,44 @@ public void tapOnElement(WebElement element) {
     jsexec.executeScript("arguments[0].click();", element);
   }
 
-  public void waitMilliseconds(int time, String methodName) {
+  public void waitMilliseconds(int time, String message) {
     try {
       Thread.sleep(time);
     } catch (InterruptedException e) {
-      PageObjectLogging.log(methodName, e.getMessage(), false);
+      PageObjectLogging.log(message, e.getMessage(), false);
+    }
+  }
+
+  /**
+   * First waits for spinner to be visible and then waits for spinner to be hidden
+   */
+  public void waitForLoadingSpinnerToFinishReloadingPage() {
+    try {
+      waitForElementByElement(loadingSpinner);
+      waitForElementPresenceByBy(By.cssSelector(".loading-overlay.hidden"));
+    } catch (TimeoutException e) {
+    }
```

Comment:
in this case, if spinner is visible, and then stays on page this method just goes on after 30 sec.